admin/server/projects.go

```diff
@@ -290,9 +293,23 @@ func (s *Server) GetProject(ctx context.Context, req *adminv1.GetProjectRequest)
 		}, nil
 	}
 
-	depl, err := s.admin.DB.FindDeployment(ctx, *proj.PrimaryDeploymentID)
-	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+	var depl *database.Deployment
+	if req.Branch != "" {
+		depls, err := s.admin.DB.FindDeploymentsForProject(ctx, proj.ID, "", req.Branch)
+		if err != nil {
+			return nil, err
+		}
+		if len(depls) == 0 {
+			return nil, status.Errorf(codes.InvalidArgument, "no deployment found for branch %q", req.Branch)
+		} else if len(depls) > 1 {
+			return nil, status.Errorf(codes.InvalidArgument, "multiple deployments found for branch %q. Recreate deployments to resolve", req.Branch)
+		}
+		depl = depls[0]
+	} else {
+		depl, err = s.admin.DB.FindDeployment(ctx, *proj.PrimaryDeploymentID)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	if !permissions.ReadProdStatus {
```

Comment:
if `depl.Environment` is not prod, I guess it should check `ReadDevStatus` instead?

(I'm not sure if it's worth refactoring, but from a semantics perspective, I guess we'll kind of treat the `ReadDev` permission more like a `ReadNonProd` permission)